grails-app/utils/org/pih/warehouse/MultipartResolver.groovy

```diff
@@ -0,0 +1,22 @@
+package org.pih.warehouse
+
+import org.pih.warehouse.core.MultipartMaxSizeException
+import org.springframework.web.multipart.MaxUploadSizeExceededException
+import org.springframework.web.multipart.MultipartHttpServletRequest
+import org.springframework.web.multipart.commons.CommonsMultipartResolver
+
+import javax.servlet.http.HttpServletRequest
+
+class MultipartResolver extends CommonsMultipartResolver {
+
+    @Override
+    MultipartHttpServletRequest resolveMultipart(HttpServletRequest request) {
+        try {
+            return super.resolveMultipart(request);
+        } catch (MaxUploadSizeExceededException e) {
+            Throwable exception = new MultipartMaxSizeException(e)
+            request.setAttribute("exception", exception)
+            throw(exception)
```

Comment:
Like I said before, I would prefer to either rethrow MaxUploadSizeExceededException or follow the similar approach i.e. returning 

Here's an example from a Grails 3 app 

https://github.com/UNCake/UNCake/blob/037c4d7c638a61a99a47d0388caf266fe3582faa/UNCake/grails-app/utils/uncake/CustomMultipartResolver.java#L4

If we cannot get it working this way then I'd suggest we go with the command object with contraints apporach for now. 
```
return new DefaultMultipartHttpServletRequest(request, new LinkedMultiValueMap<>(), new LinkedHashMap<>(), new LinkedHashMap<>());
```

There are other Spring approaches (@ExceptionHandler, @ControllerAdvice) that we might be able to take advantage of but some of them are supported in Spring 5 (we're using Spring 4), so we'll need to wait a bit until we can use them.